src/sagemaker/jumpstart/accessors.py

```diff
@@ -0,0 +1,141 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains accessors related to SageMaker JumpStart."""
+from __future__ import absolute_import
+from typing import Any, Dict, Optional
+from sagemaker.jumpstart.types import JumpStartModelHeader, JumpStartModelSpecs
+from sagemaker.jumpstart import cache
+from sagemaker.jumpstart.constants import JUMPSTART_DEFAULT_REGION_NAME
+
+
+class SageMakerSettings(object):
+    """Static class for storing the SageMaker settings."""
+
+    _parsed_sagemaker_version = ""
+
+    @staticmethod
+    def set_sagemaker_version(version: str) -> None:
+        """Set SageMaker version."""
+        SageMakerSettings._parsed_sagemaker_version = version
+
+    @staticmethod
+    def get_sagemaker_version() -> str:
+        """Return SageMaker version."""
+        return SageMakerSettings._parsed_sagemaker_version
+
+
+class JumpStartModelsCache(object):
+    """Static class for storing the JumpStart models cache."""
+
+    _cache: Optional[cache.JumpStartModelsCache] = None
+    _curr_region = JUMPSTART_DEFAULT_REGION_NAME
+
+    _cache_kwargs: Dict[str, Any] = {}
+
+    @staticmethod
+    def _validate_and_mutate_region_cache_kwargs(
+        cache_kwargs: Optional[Dict[str, Any]] = None, region: Optional[str] = None
+    ) -> Dict[str, Any]:
+        """Returns cache_kwargs with region argument removed if present.
+
+        Raises:
+            ValueError: If region in `cache_kwargs` is inconsistent with `region` argument.
+
+        Args:
+            cache_kwargs (Optional[Dict[str, Any]]): cache kwargs to validate.
+            region (str): The region to validate along with the kwargs.
+        """
+        cache_kwargs_dict = {} if cache_kwargs is None else cache_kwargs
+        assert isinstance(cache_kwargs_dict, dict)
+        if region is not None and "region" in cache_kwargs_dict:
+            if region != cache_kwargs_dict["region"]:
+                raise ValueError(
+                    f"Inconsistent region definitions: {region}, {cache_kwargs_dict['region']}"
+                )
+            del cache_kwargs_dict["region"]
```

Comment:
What's the need to delete the region key here?

Reply:
We delete this parameter since it'll be passed outside of `cache_kwargs` argument.